pkgs/development/python-modules/pre-commit-hook-ensure-sops/default.nix

```
@@ -0,0 +1,33 @@
+{ lib, buildPythonPackage, applyPatches, fetchFromGitHub, ruamel-yaml }:
+
+buildPythonPackage rec {
+  pname = "pre-commit-hook-ensure-sops";
+  version = "1.1";
+
+  src = fetchFromGitHub {
+    owner = "yuvipanda";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-8sMmHNzmYwOmHYSWoZ4rKb/2lKziFmT6ux+s+chd/Do=";
+  };
+
+  # Add the command-line entrypoint to pyproject.toml
+  patches = [ ././entrypoint.patch ];
+
+  propagatedBuildInputs = [ ruamel-yaml ];
+
+  pythonImportsCheck = [ "pre_commit_hook_ensure_sops" ];
+
+  # Test entrypoint
+  checkPhase = ''
+    $out/bin/pre-commit-hook-ensure-sops --help
+  '';
```

Overriding the `checkPhase` is probably fine in this case. Let's make sure we can still configure the pre- and post-hooks.

```suggestion
  # Test entrypoint
  checkPhase = ''
    runHook preCheck
    $out/bin/pre-commit-hook-ensure-sops --help
    runHook postCheck
  '';
```